framework/src/media/FileInputDataSource.cpp

```diff
@@ -55,24 +55,37 @@ bool FileInputDataSource::open()
 {
 	if (!mFp) {
 		setAudioType(utils::getAudioTypeFromPath(mDataPath));
-
-		switch (getAudioType()) {
-		case AUDIO_TYPE_MP3:
-		case AUDIO_TYPE_AAC:
-		case AUDIO_TYPE_OPUS:
-			setDecoder(std::make_shared<Decoder>(getChannels(), getSampleRate()));
-			break;
-		case AUDIO_TYPE_FLAC:
-			/* To be supported */
-			break;
-		default:
-			/* Don't set any decoder for unsupported formats */
-			break;
-		}
+		unsigned int channel;
+		unsigned int sample_rate;
+		audio_format_type_t pcm_format;
 
 		mFp = fopen(mDataPath.c_str(), "rb");
 		if (mFp) {
-			medvdbg("file open success\n");
+			printf("file open success\n");
+
+			if (!utils::header_parsing(mFp, getAudioType(), &channel, &sample_rate, &pcm_format)) {
+				meddbg("header parsing failed\n");
+				return false;
+			}
+			
+			setSampleRate(sample_rate);
+			setChannels(channel);
+			setPcmFormat(pcm_format);
+
+			switch (getAudioType()) {
+			case AUDIO_TYPE_MP3:
+			case AUDIO_TYPE_AAC:
+			case AUDIO_TYPE_OPUS:    
+				setDecoder(std::make_shared<Decoder>(getChannels(), getSampleRate()));    
```

Comment:
Currently decoder check audio type one more time, but here we already know audiotype..after submit this PR, improve it.